
# OUTPUT QUESTIONS

1. u have to understand the flow of execution
2. u have to do dry run with pencil 
3. whenever u see something tricky make it as imp .

In [1]:
L1 = [500, 800, 600, 200, 900]
START = 1
SUM = 0

for C in range(START, 4):  # (1,4) - 1,2,3 
    SUM = SUM + L1[C]
    print(C, ":", SUM)
    SUM = SUM + L1[0] * 10

print(SUM)

1 : 800
2 : 6400
3 : 11600
16600


1. for c= 1
 SUM = 0 + 800
 print(1, ":", 800)
 SUM = 800 + 500 * 10 = 800+5000=5800

2. for c= 2
 sum = 5800+600 = 6400
 print(2, ":",6400)
 sum = 6400+5000 = 11400    

In [2]:
a = "Year 2022 at All the best"
a = a.split('2')
b = a[0] + "." + a[1] + "." + a[3]
print(b)

Year .0. at All the best


In [3]:
my_dict = {"name": "Aman", "age": 26}
my_dict['age'] = 27
my_dict['address'] = "Delhi"
print(my_dict.items())

dict_items([('name', 'Aman'), ('age', 27), ('address', 'Delhi')])


In [4]:
for Name in ['Jayes', 'Ramya', 'Taruna', 'Suraj']:
    print(Name)
    if Name[0] == 'T':
        break
    else:
        print('Finished!')
print('Got it!')

Jayes
Finished!
Ramya
Finished!
Taruna
Got it!


In [ ]:
def Diff(N1, N2): # n1= 32 , n2= 54
    if N1 > N2: # 32>54
        return N1 - N2
    else:
        return N2 - N1 # 54-32 = 22

NUM = [10, 23, 14, 54, 32]
for CNT in range(4, 0, -1):
    A = NUM[CNT]
    B = NUM[CNT - 1]

    print(Diff(A, B), '#', end=' ')


# cnt = 4 3 2 1
cnt= 4
a=num[4] = 32
b=num[4-1=3]= 54
cnt =3
a=num[3] = 54
b=num[3-1=2]= 14


In [9]:
tuple1 = (11, 22, 33, 44, 55, 66)
list1 = list(tuple1)
new_list = [] # empty list 
for i in list1:
    if i % 2 == 0: # 22%2==0
        new_list.append(i)

new_tuple = tuple(new_list)
print(new_tuple)

(22, 44, 66)


1. list1=[11,22,33,44,55,66]
2. 11 does not get appended 
3. [22] bcz 22%2==0
4. 33 does not get append 
5. [22,44] bcz 44%2==0
6. 55 does not get append
7. [22,44,66]
8. new_tuple = tuple([22,44,66])
9. new_tuple = (22,44,66)

In [10]:
p = 5 # p is global 

def sum(q, r=2): # q=10 r= 5
    global p # global is a keyword which is used to make any local variable into the global variable
    p = r + q ** 2 
    print(p, end='#')

a = 10
b = 5
sum(a, b) # sum(10,5)
sum(r=5, q=1)

105#6#

1. global variable and local variable and gloabl keyword
2. p = 5 + 10 ** 2 = 5+100=105
3. 105#
4. sum(r=5, q=1)
5. p = 5 + 1 ** 2 = 5+1=6
6. 105#6#

In [23]:
s = "welcome2cs"
n = len(s)
m = "" 

for i in range(0, n): # (0,10)= 0 1 2 3 4 5 6 7 8 9 
    if s[i] >= 'a' and s[i] <= 'm':
        m = m + s[i].upper()
    elif s[i] >= 'n' and s[i] <= 'z':
        m = m + s[i-1]
    elif s[i].isupper():
        m = m + s[i].lower()
    else:
        m = m +'&'

print(m)

sELCcME&Cc


1. w>n and w<z
2. m = "" + "s"="s"
3. m = "s"+"E"="sE"

               

In [16]:
# and(*) or(+) not(~) - logiacl operator 
print(1 and 0)
print(0 and 0)
print(0 and 1)
print(1 and 1)
print(0 or 0)
print(1 or 0)

0
0
0
1
0
1


In [6]:
import random 
x=random.random()
print(x)
# 0 and 1 both are excluded

0.03557405586723361


In [18]:
import random 
x=random.randint(1,5)
print(x)
# ending value excluded 

3


In [192]:
import random 
x=random.randrange(1,5)
y=random.randrange(10)
print(y)
print(x)
# ending value excluded

4


In [221]:
import random 
x=["aman","sofia","bhavya","adiba"]
random.choice(x)

'aman'

In [222]:
def ChangeVal(M, N):
    for i in range(N):
        if M[i] % 5 == 0:
            M[i] //= 5
        if M[i] % 3 == 0:
            M[i] //= 3

L = [25, 8, 75, 12]

ChangeVal(L, 4)

for i in L:
    print(i, end="#")

5#8#5#4#

In [223]:
if 15%5==0:
    print("hoga")
if 15%3==0:
    print("hoga")

hoga
hoga
